### ISSUE TITLE
Ability to treat comments as code

### ISSUE THREAD

**Issue by timothycrosley** [timothycrosley/isort#1357]
isort should allow treating all single line comments (or chosen single line comments) as code if that is the end users preference. 

For example:

```
import b

# these are special imports that have to do with installing X plugin
import c
import a
```

by default isort would turn this into:

```
# these are special imports that have to do with installing X plugin
import a
import b
import c
```

with the new `treat_all_comments_as_code` option turned on it would become:

```
import b

# these are special imports that have to do with installing X plugin
import a
import c
```

**Comment by timothycrosley** [comment]
This supersedes: https://github.com/timothycrosley/isort/issues/1338

**Comment by timothycrosley** [comment]
This has been released with version 5.3.0 of isort

**Comment by yasirroni** [comment]
It seems that there is a bug in this mode?

I'm using this setting and it seems to not working.

```
  # sort import using isort
  - repo: https://github.com/pycqa/isort
    rev: 5.8.0
    hooks:
      - id: isort
        args: [
        '--treat-comment-as-code',
        '"# %%"',
        '--'
        ]
```
